packages/frontend/src/stdlib/analyses/schema_erd.tsx

```diff
@@ -0,0 +1,263 @@
+    };
+
+    const title = () => props.title ?? "Entity-Relationship Diagram";
+    const header = () => (
+        <>
+            <DownloadSVGButton svg={svgRef()} tooltip={`Export the ${title()} as SVG`} size={16} />
+        </>
+    );
+
+    return (
+        <div class="graph-visualization-container">
+            <Foldable title={title()} header={header()}>
+                <FormGroup compact>
+                    <SelectField
+                        label="Direction"
+                        value={props.content.direction ?? Direction.Vertical}
+                        onChange={(evt) => {
+                            props.changeContent((content) => {
+                                content.direction = evt.currentTarget.value as Direction;
+                            });
+                        }}
+                    >
+                        <option value={Direction.Horizontal}>{"Horizontal"}</option>
+                        <option value={Direction.Vertical}>{"Vertical"}</option>
+                    </SelectField>
+                </FormGroup>
+            </Foldable>
+            <div class="graph-visualization">
+                <Show when={svgString()}>
+                    {(svg) => (
+                        <div
+                            ref={(el) => {
+                                const svgEl = el.querySelector("svg");
+                                if (svgEl) {
+                                    setSvgRef(svgEl);
+                                }
+                            }}
+                            innerHTML={svg()}
+                        />
+                    )}
+                </Show>
+            </div>
+        </div>
+    );
+}
+
+/** Convert a schema model into an ERD-style Graphviz graph using HTML-like labels. */
+export function schemaToERD(model: DblModel): Viz.Graph {
+    const entityType: ObType = { tag: "Basic", content: "Entity" };
+    const attrType: MorType = { tag: "Basic", content: "Attr" };
+
+    const entities = model.obGeneratorsWithType(entityType);
+    const nodes: Required<Viz.Graph>["nodes"] = [];
+
+    // Collect all mappings to know which entities they point to
+    const mappingsByEntity = new Map<
+        string,
+        Array<{ id: string; name: string; targetEntity: string }>
+    >();
+    for (const morId of model.morGenerators()) {
+        const mor = model.morPresentation(morId);
+        if (!mor) {
+            continue;
+        }
+
+        const isMappingType =
+            mor.morType.tag === "Hom" &&
+            mor.morType.content.tag === "Basic" &&
+            mor.morType.content.content === "Entity";
+
+        if (
+            isMappingType &&
+            mor.dom.tag === "Basic" &&
+            mor.cod.tag === "Basic" &&
+            entities.includes(mor.dom.content) &&
+            entities.includes(mor.cod.content)
+        ) {
+            const mappingName = mor.label?.every((seg) => seg !== "") ? mor.label.join(".") : morId;
```

Comment:
The `else` clause doesn't seem right: `morId` is a UUID and thus should never be displayed. In this case, you could return `""` or something like `"unnamed"`.